src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -156,12 +156,7 @@ private void GetNextSpan()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Advance(int count)
         {
-            if (count == 0)
-            {
-                return;
-            }
-
-            if (count < 0 || End)
+            if (count < 0)
```

Comment:
Please note the changes here. I tried to trim Advance as much as possible.

Do we actually need the Consumed property?

cc @JeremyKuhne